contracts/libraries/BytesLib.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+library BytesLib {
+    /**************************************
+     *              ERRORS                *
+     **************************************/
+    error OutOfBounds();
+    error InvalidBytes();
+    error InvalidStart();
+
+    /**************************************
+     *              FUNCTIONS              *
+     **************************************/
+
+    /**
+     * @notice Reads a uint32 from a bytes array at a given start index
+     * @param _bytes The bytes array to convert
+     * @param _start The start index of the uint32
+     * @return result The uint32 result
+     */
+    function toUint32(bytes memory _bytes, uint256 _start) internal pure returns (uint32 result) {
+        if (_bytes.length < _start + 4) {
+            revert OutOfBounds();
+        }
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            result := mload(add(add(_bytes, 0x4), _start))
+        }
+    }
+
+    /**
+     * @notice Reads a uint256 from a bytes array at a given start index
+     * @param _bytes The bytes array to convert
+     * @param _start The start index of the uint256
+     * @return result The uint256 result
+     */
+    function toUint256(bytes memory _bytes, uint256 _start) internal pure returns (uint256 result) {
+        if (_bytes.length < _start + 32) {
+            revert OutOfBounds();
+        }
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            result := mload(add(add(_bytes, 0x20), _start))
+        }
+    }
+
+    /**
+     * @notice Reads a bytes32 from a bytes array at a given start index
+     * @param _bytes The bytes array to convert
+     * @param _start The start index of the bytes32
+     * @return result The bytes32 result
+     */
+    function toBytes32(bytes memory _bytes, uint256 _start) internal pure returns (bytes32 result) {
+        if (_bytes.length < _start + 32) {
+            revert OutOfBounds();
+        }
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            result := mload(add(add(_bytes, 0x20), _start))
+        }
+    }
+
+    /**
+     * @notice Reads a bytes array from a bytes array at a given start index and length
+     * @param _bytes The bytes array to convert
+     * @param _start The start index of the bytes array
+     * @param _end The end index of the bytes array
+     * @return result The bytes array result
+     */
+    function slice(bytes memory _bytes, uint256 _start, uint256 _end) internal pure returns (bytes memory result) {
```

Comment:
Can we import OZ's function instead of homerolling? https://github.com/OpenZeppelin/openzeppelin-contracts/blob/1cf13771092c83a060eaef0f8809493fb4c04eb1/contracts/utils/Bytes.sol#L80-L98